apps/service_persist/lib/persist/loader.ex

```diff
@@ -24,11 +26,18 @@ defmodule Persist.Loader do
     Process.flag(:trap_exit, true)
     %Load.Persist{} = load = Keyword.fetch!(init_arg, :load)
 
-    with {:ok, writer_pid} <- start_writer(load),
-         writer_function <- fn msgs -> writer().write(writer_pid, msgs, schema: load.schema) end,
-         {:ok, broadway_pid} <- start_broadway(load, writer_function) do
+    with {:ok, transform} when not is_nil(transform) <- Transformations.get(load.dataset_id),
+         {:ok, dictionary} <- transform_dictionary(transform),
+         {:ok, writer_pid} <- start_writer(load, dictionary),
+         writer_function <- fn msgs ->
+           writer().write(writer_pid, msgs, dictionary: dictionary)
+         end,
+         {:ok, broadway_pid} <- start_broadway(load, transform, writer_function) do
       {:ok, %{writer_pid: writer_pid, broadway_pid: broadway_pid}}
     else
+      {:ok, nil} ->
```

Comment:
since you said there always needs to be a transform, should it assume a default `:noop` transform if it doesn't find one?

Reply:
I think stopping the process is best, because if now transform is found then we also don't know the dictionary and create the table in presto.